letta/agent.py

```diff
@@ -1612,18 +1618,29 @@ def count_tokens(self) -> int:
 
 def save_agent(agent: Agent, ms: MetadataStore):
     """Save agent to metadata store"""
-
     agent.update_state()
     agent_state = agent.agent_state
     assert isinstance(agent_state.memory, Memory), f"Memory is not a Memory object: {type(agent_state.memory)}"
 
     # TODO: move this to agent manager
+    # TODO: Completely strip out metadata
     # convert to persisted model
-    persisted_agent_state = agent.agent_state.to_persisted_agent_state()
-    if ms.get_agent(agent_id=persisted_agent_state.id):
-        ms.update_agent(persisted_agent_state)
-    else:
-        ms.create_agent(persisted_agent_state)
+    agent_manager = AgentManager()
+    update_agent = UpdateAgent(
+        name=agent_state.name,
+        tool_ids=[t.id for t in agent_state.tools],
+        source_ids=[s.id for s in agent_state.sources],
+        block_ids=[b.id for b in agent_state.memory.blocks],
+        tags=agent_state.tags,
+        system=agent_state.system,
+        tool_rules=agent_state.tool_rules,
+        llm_config=agent_state.llm_config,
+        embedding_config=agent_state.embedding_config,
+        message_ids=agent_state.message_ids,
+        description=agent_state.description,
+        metadata_=agent_state.metadata_,
+    )
+    agent_manager.update_agent(agent_id=agent_state.id, agent_update=update_agent)
```

Comment:
will this also save the blocks? if so, maybe make a note of it?

Reply:
it requires that the passed in block ids already exist